components/shared/button.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <v-btn
+    v-bind="$attrs"
+    :color="icon ? undefined : inversed ? textColor : parsedColor"
+    :style="
+      !icon &&
+      parsedVariant === 'contained' && {
+        color: inversed ? parsedColor : textColor,
+      }
+    "
+    :outlined="parsedVariant === 'outlined'"
+    :text="parsedVariant === 'text'"
+    :icon="icon"
+    @click="$emit('click', $event)"
+  >
+    <slot />
+  </v-btn>
+</template>
+
+<script>
+export default {
+  props: {
+    color: {
+      type: String,
+      default: 'primary',
+    },
+    variant: {
+      type: String,
+      default: 'contained',
+    },
+    inversed: {
+      type: Boolean,
+      default: false,
+    },
+    icon: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      colorToTextColorMap: {
+        dark: {
+          primary: '#121212',
+          secondary: '#121212',
+          accent: '#fff',
+          success: '#082218',
+          warning: '#2A150D',
+          error: '#0e0000',
+          info: '#050709',
+        },
+        light: {
+          primary: '#fff',
+          secondary: '#fff',
+          accent: '#121212',
+          success: '#082218',
+          warning: '#2A150D',
+          error: '#0e0000',
+          info: '#050709',
+        },
+      },
+    }
+  },
+  computed: {
+    parsedVariant: function () {
+      return this.isAcceptedVariant(this.variant) ? this.variant : 'contained'
+    },
+    parsedColor: function () {
+      const color = this.isAcceptedColor(this.color) ? this.color : 'primary'
```

Comment:
What is your opinion of instead doing something like this instead of `isAcceptedColor` method? Might be a little bit cleaner, in my opinion.

`['primary', 'secondary', 'accent', 'info', 'success', 'warning', 'error'].includes(this.color)`

Where the array could be stored as `this.acceptedColors`